### PR TITLE
Added json for ollama script

### LINKED ISSUE #202: ollama script not on the website
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

Converting this [discussion](https://github.com/community-scripts/ProxmoxVE/discussions/201) into a issue

> it's available to start / install from :
bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/ollama.sh)"
but missing in the website :

### What settings are you currently utilizing?

Default Settings

### Which Linux distribution are you employing?

Ubuntu 22.04

### If relevant, including screenshots or a code block can be helpful in clarifying the issue.

No JSON exists.

### Please provide detailed steps to reproduce the issue.

Search the website

### PR DESCRIPTION
**Pull request by newzealandpaul**
> [!NOTE]
> We are meticulous when it comes to merging code into the main branch, so please understand that we may reject pull requests that do not meet the project's standards. It's never personal. Also, game-related scripts have a lower chance of being merged.

## Description

ollama did not have a json file and so did not appear on the website.

Fixes #202 

## Type of change
Please check the relevant option(s):

- [x] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [] Testing performed (I have tested my changes, ensuring everything works as expected)
- [x] Documentation updated (I have updated any relevant documentation)

## Related Pull Requests / Discussions

https://github.com/community-scripts/ProxmoxVE/discussions/201

### CODE REVIEW

File: json/ollama.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            }
+        }
+    ],
+    "default_credentials": {
+        "username": null,
+        "password": null
+    },
+    "notes": [
+        {
+            "text": "Due to the size of the models, 24GB of storage is required at a minimum.",
```
Comment (havardthom) [inline json/ollama.json:35]:
```suggestion
```
The script does not install any models, It is the application and dependencies that requires this size

**Review by havardthom — COMMENTED** [review]
Not sure we should add this. The script was created to have Ollama as separate LXC and support intel iGPU passtrough, but it was discovered that Ollama does not support intel iGPU yet, we need to wait for https://github.com/ollama/ollama/pull/5593

The script builds Ollama from source which can take 20-40 minutes. It does not consider Nvidia or AMD support in the build, so anyone that wants to utilize those need to rebuild with their own drivers/dependencies.

For now, it is better for everyone to use the Ollama option in Open WebUI script instead, since that installs precompiled binary instead, which also has runners for Nvidia and AMD, so the user only has to passthrough GPU and install drivers for it to work.

If do add this, we need to have some more notes:
- `Installation can take 20-40 minutes because the script builds Ollama from source`

- `Intel iGPU hardware acceleration is not yet supported in Ollama, see current status in `https://github.com/ollama/ollama/pull/5593``

- `For Nvidia or AMD hardware acceleration, Ollama must be rebuilt with appropriate drivers and dependencies`

- `Ollama can also be installed as an option in the Open WebUI helper script `https://community-scripts.github.io/ProxmoxVE/scripts?id=openwebui``

File: json/ollama.json (hunk 1 above)
Comment (newzealandpaul) [reply json/ollama.json:35]: I will add a note saying additional storage is required for models. 

But why does the script allocate 24GB of disk space, when only 11GB is used on a fresh install?

**Comment by newzealandpaul** [comment]
Added note saying that Ollama LCX disk may need to be resized. Some of the models are larger than 40gb.

File: json/ollama.json (hunk 1 above)
Comment (havardthom) [reply json/ollama.json]: I think mine went up to 19 gb at one point, but i was testing many different things so maybe 24gb is not actually needed

**Comment by newzealandpaul** [comment]
> Not sure we should add this. The script was created to have Ollama as separate LXC and support intel iGPU passtrough, but it was discovered that Ollama does not support intel iGPU yet, we need to wait for https://github.com/ollama/ollama/pull/5593

I think we either should merge this, or delete the script.

This brings up a discussion on what our policy should be on retiring scripts that people may rely on @community-scripts/contributor.

**Comment by havardthom** [comment]
I think the script is viable when Ollama merges https://github.com/ollama/ollama/pull/5593, but currently the Ollama install in Open WebUI script is a better option. Perhaps the script can be tagged as a work in progress (WIP) somehow

**Comment by MickLesk** [comment]
Do we agree that we should put this pull request on hold for the moment?

**Comment by remz1337** [comment]
I don't think tteck would've accepted that. I would say let's hold off for now until we have something more stable.

**Pull request closed without merging**